test/e2e/crud.go

```diff
@@ -44,49 +50,128 @@ var _ = Describe("Federated", func() {
 	// resources is performed in the event of test failure before
 	// controllers are shutdown.
 
+	validatedUnlabeledNotDeleted := false
 	for key := range typeConfigFixtures {
 		typeConfigName := key
 		fixture := typeConfigFixtures[key]
 		Describe(fmt.Sprintf("%q", typeConfigName), func() {
 			It("should be created, read, updated and deleted successfully", func() {
-				// Lookup the type config from the api
-				client, err := genericclient.New(f.KubeConfig())
-				if err != nil {
-					tl.Fatalf("Error initializing dynamic client: %v", err)
+				typeConfig, testObjectsFunc := getCrudTestInput(f, tl, typeConfigName, fixture)
+				crudTester, targetObject, overrides := initCrudTest(f, tl, typeConfig, testObjectsFunc)
+				crudTester.CheckLifecycle(targetObject, overrides)
+			})
+
+			// Unlabeled resource handling behavior should not vary between
+			// types, so testing a single type is sufficient.  Picking a
+			// namespaced type minimizes the impact of teardown failure.
+			if typeConfigName != "configmaps" || validatedUnlabeledNotDeleted {
```

Comment:
Why is the additional flag required? Isn't there a single fixture per type as of now.

Reply:
There is no need to validate the unlabeled handling for more than a single type, since the behavior is common across all types.

Reply:
What I meant was, is a check like below not sufficient? (why is `validatedUnlabeledNotDeleted` needed?) :

```
if typeConfigName != "configmaps" {
    return
}
```

Reply:
Am merging this as this is anyways non-functional and don't see a point wasting time on it.

Reply:
Ah, gotcha.  I recall now that the flag was an attempt to run the test on any type before I realized that a namespaced type was preferable.  Done.